url.bs

```diff
@@ -1541,8 +1587,19 @@ string <var>input</var>, optionally with a <a>base URL</a> <var>base</var>, opti
        <li><p><var>url</var> <a>is special</a> and <a>c</a> is "<code>\</code>"
       </ul>
 
-      <p>then decrease <var>pointer</var> by the number of code points in <var>buffer</var> plus
-      one, set <var>buffer</var> to the empty string, and set <var>state</var> to <a>host state</a>.
+      <p>then run these substeps:
+
+      <ol>
+       <li><p>If <var>@ flag</var> is set and <var>buffer</var> is the empty string,
+       <a>syntax violation</a>, return failure.
+       <!-- No URLs with userinfo, but without host. For special URLs it would also not be
+            idempotent:
+            https://@/example.org/ -> https:///example.org/ -> https://example.org/ -->
+
+       <li><p>Decrease <var>pointer</var> by the number of code points in <var>buffer</var> plus
```

Comment:
I left this as-is since the parser generally groups steps that can reasonably be grouped.